### PR TITLE
Finally fixed Staff Dashboard and NavBar Styling

### PR DESCRIPTION
**Pull request by didvi**
## New Staff Dashboard

I completely re-did the staff dashboard to support more responsive styling. There are some tweaks that need to be done to colors to match the Figma, otherwise everything still works the same.

I also added two important components: `Main` and `EnhancedTable`. `Main` should be used as the base component for every screen, as it has the NavBar in it and corresponding styling. All props are forwarded to the component that is specified. `EnhancedTable` is a beautiful component for any type of dashboard, and it supports filtering of columns right off the bat.

**Annoying Bugs I Fixed**
-  Navbar used to be junk and had to be imported into each page and then styled. Now is in `Main` 
- Table styling was all jank scss that was converted to jss. This is no longer, which also means the styling doesn't mess up components that it's tangentially related to.
- Staff Dashboard can filter by columns
- Search bar looks more visible and beautiful
- Plus button is no longer slightly off center and bad looking
- Fixed error with participant card not showing the correct number of case notes.

**Bugs that still need fixing**
- There is some text from the rails side that shows up in the back of every screen @wilsonw926 is looking into this
- The header for the Staff Dashboard is different than the figma (although I kind of like it)
- The participant show page styling may be off once we add content that spans more than `100vh`, since the height of the background div is specified as `100vh`.
- Studio Assessments Dashboard header is different than Staff Dashboard
- I have not tried filtering with different statuses for a participant.

### Related PRs
@kyleqhua I added some tweaks to your studio assessment dashboard to get the navbar working, which broke a bit of the styling. I will fix this once your PR is merged in. This also might raise some conflicts when merging but I shall deal with it.
I also added a component Enhanced Table which you can use for your dashboard as well if you so choose. It generalizes very well.

### Screenshots

Before:
image

image

After:
image

image

CC: @didvi

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def create_participants
       last_name: Faker::Name.last_name,
       password: 'password',
       password_confirmation: 'password',
-      user_type: 0
+      user_type: rand(2)
```
Comment (wilsonw926) [inline db/seeds.rb:56]: Was this supposed to be changed?

File: app/javascript/components/Main/index.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        component="a"
+        disableFocusRipple
+        disableTouchRipple
+        className={classes.navBarItem}
+        onClick={() => window.location.href = '/admin'}
+        >
+        <ExitToAppIcon />
+        <div className = {classes.navText} > Admin View </div>
+        </ListItem>
+
```
Comment (wilsonw926) [inline app/javascript/components/Main/index.js:62]: rm line

File: app/javascript/components/EnhancedTable/index.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+EnhancedTableHead.propTypes = {
+  classes: PropTypes.object.isRequired,
+  onRequestSort: PropTypes.func.isRequired,
+  order: PropTypes.oneOf(['asc', 'desc']).isRequired,
+  orderBy: PropTypes.string.isRequired,
+  headCells: PropTypes.array.isRequired,
+};
+function EnhancedTable(props) {
```
Comment (wilsonw926) [inline app/javascript/components/EnhancedTable/index.js:91]: add line

File: db/seeds.rb (hunk 1 above)
Comment (didvi) [reply db/seeds.rb:56]: no

File: app/javascript/components/Main/styles.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      flexShrink: 0,
+    },
+    drawerPaper: {
+      width: drawerWidth,
+      backgroundColor: "#29313C",
+    },
+    // necessary for content to be below app bar
+    content: {
+      flexGrow: 1,
+      backgroundColor: 'rgba(210, 220, 225, 0.63)',
```
Comment (kyleqhua) [inline app/javascript/components/Main/styles.js:21]: move color to theme.js?

**Pull request merged**